src/webgl/loading.js

```diff
@@ -213,7 +287,20 @@ function parseObj(model, lines) {
     const tokens = lines[line].trim().split(/\b\s+/);
 
     if (tokens.length > 0) {
-      if (tokens[0] === 'v' || tokens[0] === 'vn') {
+      if (tokens[0] === 'usemtl') {
+        // Switch to a new material
+        currentMaterial = tokens[1];
+        if (currentMaterial && materials[currentMaterial]) {
+          const diffuseColor = materials[currentMaterial].diffuseColor;
+          model.vertexColors.push([
```

Comment:
I think we want to add to the diffuse colors *for each vertex*, where currently this is adding to the array once for each `usemtl` line. So maybe that means just setting the `currentMaterial` in this `if` branch, and in the branch below where we add to `loadedVerts`, add to `vertexColors` based on the material.